assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.test.js

```diff
@@ -121,6 +128,116 @@ describe( 'WidgetAreaRenderer', () => {
 		} );
 	} );
 
+	it( 'should only render widgets the user has access to in a view-only viewContext', async () => {
+		createWidgets( registry, areaName, [
+			{
+				Component: WidgetComponent,
+				modules: 'search-console',
+				slug: 'one',
+				width: WIDGET_WIDTHS.FULL,
+			},
+			{
+				Component: WidgetComponent,
+				modules: 'search-console',
+				slug: 'two',
+				width: WIDGET_WIDTHS.FULL,
+			},
+			{
+				Component: () => <div>AdSense is here</div>,
+				modules: 'adsense',
+				slug: 'three',
+				width: WIDGET_WIDTHS.FULL,
+			},
+		] );
+
+		// Add our dashboard sharing capabilities into state.
+		fetchMock.getOnce(
+			/^\/google-site-kit\/v1\/core\/modules\/data\/list/,
+			{
+				body: FIXTURES,
+				status: 200,
+			}
+		);
```

Comment:
Do we need to use fetchMock here? Generally we reserve that for cases where we're specifically testing a resolver or it isn't possible to receive otherwise. If not, let's use `provideModules` instead.

Reply:
Maybe not, I just copied this from the only similar test for this stuff, but fair enough. 👍🏻